File: 2018/1.py
series = [int(n) for n in open("1.input").read().split()]
part1 = sum(series)

def part2(series):
    seen = set()
    i = 0
    value = 0
    while True:
        value += series[i]
        if value in seen:
            return value
        seen.add(value)
        i = (i + 1)%len(series)

print("part1:", part1)
print("part2:", part2(series))

File: 2019/1.hs
-- input = readFile "aoc1.input" >>= \str -> return (lines str)
-- input = fmap lines $ readFile "aoc1.input"

input :: IO [Int]
input = do
  str <- readFile "1.input"
  let ls = map read . lines $ str
  return ls

fuel :: Int -> Int
fuel m = max 0 $ floor (fromIntegral(m) / 3) - 2

rocket_fuel :: Int -> Int
rocket_fuel m = sum $ drop 1 $
                takeWhile (\x -> x > 0) overhead where
  overhead = m : [fuel y | y <- overhead]

part1 = do
  modules <- input
  let fueled = fmap fuel modules
  return (sum fueled)

part2 = do
  modules <- input
  let fueled = fmap rocket_fuel modules
  return (sum fueled)

main = part1 >>= \a -> print a
    >> part2 >>= \b -> print b
